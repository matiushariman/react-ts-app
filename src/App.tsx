import * as React from 'react';

import GlobalStyle from 'global-styles';

function App(): React.ReactElement {
  return (
    <>
      <h1>This is home page</h1>
      <GlobalStyle />
    </>
  );
}

export default App;
