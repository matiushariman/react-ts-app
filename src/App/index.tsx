import * as React from 'react';
import { Routes, Route, Link } from 'react-router-dom';

import HomePage from 'Pages/HomePage';
import SecondaryPage from 'Pages/SecondaryPage';
import GlobalStyle from 'global-styles';

function App(): React.ReactElement {
  return (
    <>
      <nav>
        <Link to="/">HomePage</Link>
        <Link to="secondary">SecondaryPage</Link>
      </nav>
      <Routes>
        <Route path="/" element={<HomePage />} />
        <Route path="/secondary" element={<SecondaryPage />} />
      </Routes>
      <GlobalStyle />
    </>
  );
}

export default App;
