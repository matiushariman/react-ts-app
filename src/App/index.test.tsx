import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { cleanup, render, fireEvent } from '@testing-library/react';

import App from '.';

afterEach(cleanup);

test('renders the component', () => {
  const { container } = render(
    <MemoryRouter>
      <App />
    </MemoryRouter>,
  );

  expect(container).toMatchSnapshot();
});

test('render HomePage on the first load', () => {
  const { container } = render(
    <MemoryRouter>
      <App />
    </MemoryRouter>,
  );

  expect(container.innerHTML).toMatch('HomePage');
});

test('clicking on Link redirect to the correct page', () => {
  const { container, getByText } = render(
    <MemoryRouter>
      <App />
    </MemoryRouter>,
  );

  fireEvent.click(getByText(/secondary/i));

  expect(container.innerHTML).toMatch('SecondaryPage');
});
