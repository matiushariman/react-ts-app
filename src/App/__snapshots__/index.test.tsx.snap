// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<div>
  <nav>
    <a
      href="/"
    >
      HomePage
    </a>
    <a
      href="/secondary"
    >
      SecondaryPage
    </a>
  </nav>
  <h1
    class="sc-AxjAm hgJWxN"
  >
    HomePage
  </h1>
</div>
`;
